### DISCUSSION TITLE
Need to stop external calls

### DISCUSSION THREAD

**Discussion by akshaykant** [blockscout/frontend#1211] [Q&A]
**Description:**

We are running a private blockchain using the Besu client on a VM with an internal network setup. And we are running blockscout connecting to this besu node.
We have observed continuous DNS lookups to **raw.githubusercontent.com** and **api.coingecko.com**, which I believe are unnecessary for this setup. As we do not need any external call.

**Expected Behavior:**

Given that my nodes are all on an internal network and  not relying on external configurations/scripts along with blockscout, I would expect no external DNS lookups, especially continuous ones.

What could be the expected resolution for this.

**Comment by tom2drum** [comment]
Please provide your set of ENV variables with which you are running the frontend app. Also, could you give examples of external URL calls that you see in your logs? At the moment, for me, this issue is a little vague.

**Reply by akshaykant** [reply]
These are the two URLs, it is trying to resolve as part of DNS lookup 
raw.githubusercontent.com and api.coingecko.com

And using basic setup to run postgres and docker image of blockscout. Not sure if this needs to be raised in blockscout channel and not frontend

**Reply by tom2drum** [reply]
In general, URLs of external resources are not hardcoded in our code, they are defined in ENV variables that you pass to your containers. The only exception is `NEXT_PUBLIC_API_SPEC_URL` which starting from the new release [v1.12.0](https://github.com/blockscout/frontend/releases/tag/v1.12.0-alpha) will have the default value `https://raw.githubusercontent.com/blockscout/blockscout-api-v2-swagger/main/swagger.yaml`. You can overwrite this behavior by providing a URL to your internal swagger file. But please let us know if you don't want to have a REST API page on your instance for some reason, then we will make the necessary adjustments in the release.

Though, I am not aware of any hardcoded URL in the main blockscout repo. @nikitosing can you advise here, please?

**Reply by akshaykant** [reply]
> In general, URLs of external resources are not hardcoded in our code, they are defined in ENV variables that you pass to your containers. The only exception is `NEXT_PUBLIC_API_SPEC_URL` which starting from the new release [v1.12.0](https://github.com/blockscout/frontend/releases/tag/v1.12.0-alpha) will have the default value `https://raw.githubusercontent.com/blockscout/blockscout-api-v2-swagger/main/swagger.yaml`. You can overwrite this behavior by providing a URL to your internal swagger file. But please let us know if you don't want to have a REST API page on your instance for some reason, then we will make the necessary adjustments in the release.
> 
> Though, I am not aware of any hardcoded URL in the main blockscout repo. @nikitosing can you advise here, please?

This is the command we are using to run blockscout on the besu network, which is making external DNS calls.

``podman run -d --name blockscout --network=test --log-driver none -e PORT=4000 -e DATABASE_URL=ecto://postgres:postgres@blockscoutpostgres/postgres?ssl=false -e POSTGRES_PASSWORD=test -e POSTGRES_USER=test -e NETWORK=POW -e SUBNETWORK=Test4441 -e CHAIN_ID=1337 -e COIN=POW -e BLOCK_TRANSFORMER=base -e ETHEREUM_JSONRPC_VARIANT=geth -e ETHEREUM_JSONRPC_TRANSPORT=http -e ETHEREUM_JSONRPC_HTTP_URL=http://ibft-network1-node1-1:8545 -e ETHEREUM_JSONRPC_TRACE_URL=http://ibft-network1-node1-1:8545 -e API_PATH=/blockscout-api/ -e BLOCKSCOUT_HOST=`hostname` -p 4000:4000 localhost/consensys-blockscout /usr/local/bin/mix phx.server``
